### PR TITLE
Add getDynamicValue for SizeMul and SizeDiv nodes.

### PR DESCRIPTION
**Pull request by vanbasten23**

### CODE REVIEW

**Review by JackCaoG — APPROVED** [review]
Thanks!

File: torch_xla/csrc/ops/dynamic_ir.cpp [hunk 1]
```diff
@@ -121,6 +129,14 @@ SizeDiv::SizeDiv(torch::lazy::Value a, torch::lazy::Value b)
   upper_bound_ = dim_node_0->getStaticValue() / dim_node_1->getStaticValue();
 };
 
+int64_t SizeDiv::getDynamicValue() const {
+  const torch::lazy::DimensionNode* dim_node_0 = DimCast(operand(0));
+  const torch::lazy::DimensionNode* dim_node_1 = DimCast(operand(1));
+  XLA_CHECK(dim_node_0);
+  XLA_CHECK(dim_node_1);
+  return dim_node_0->getDynamicValue() / dim_node_1->getDynamicValue();
```
Comment (miladm) [inline torch_xla/csrc/ops/dynamic_ir.cpp:137]: Why do we not have the following check?

```
XLA_CHECK(dim_node_1->getDynamicValue() != 0)
      << "Can't divide a dimension by zero";
```

CC @vanbasten23 @JackCaoG

File: test/cpp/test_ir.cpp [hunk 2]
```diff
@@ -279,5 +306,30 @@ TEST_F(IrTest, TestSizeDivNode) {
...
+      torch::lazy::MakeNode<SizeNode>(node_with_dynamism, 1);
+
+  torch::lazy::NodePtr node_div = torch::lazy::MakeNode<SizeDiv>(
+      torch::lazy::Value(size_node_nonzero_0, 0),
+      torch::lazy::Value(size_node_nonzero_1, 0));
+
+  std::shared_ptr<torch::lazy::DimensionNode> dim_node_div =
+      std::dynamic_pointer_cast<torch::lazy::DimensionNode>(node_div);
+  EXPECT_EQ(dim_node_div->getStaticValue(), 50);
+  EXPECT_EQ(dim_node_div->getDynamicValue(), 0);
```
Comment (miladm) [inline test/cpp/test_ir.cpp:331]: This test doesn't check for divide by zero. How do we address that?

@vanbasten23

**Review by miladm — COMMENTED** [review]
Reviewed this PR. Thanks @vanbasten23. 

I added a couple of comments to the PR. Since it's already merged, feel free to open a new PR to address comments.

File: torch_xla/csrc/ops/dynamic_ir.cpp (hunk 1 above)
Comment (JackCaoG) [reply torch_xla/csrc/ops/dynamic_ir.cpp:137]: yea, I guess we can have this check

Comment (miladm) [reply torch_xla/csrc/ops/dynamic_ir.cpp:137]: @vanbasten23 can you please add the PR?

Comment (vanbasten23) [reply torch_xla/csrc/ops/dynamic_ir.cpp:137]: Thanks for pointing it out. Working on it.

File: test/cpp/test_ir.cpp (hunk 2 above)
Comment (vanbasten23) [reply test/cpp/test_ir.cpp:331]: Created a test: https://github.com/pytorch/xla/pull/4051

**Comment by vanbasten23** [comment]
> Reviewed this PR. Thanks @vanbasten23.
> 
> I added a couple of comments to the PR. Since it's already merged, feel free to open a new PR to address comments.

Thanks for reviewing. Addressed it in https://github.com/pytorch/xla/pull/4051.

**Pull request merged**